rust/connlib/tunnel/src/client.rs

```diff
@@ -1459,35 +1452,27 @@ impl ClientState {
             return;
         }
 
-        let added = match &new_resource {
-            Resource::Dns(dns) => self.stub_resolver.add_resource(dns.id, dns.address.clone()),
-            Resource::Cidr(cidr) => {
-                let existing = self.active_cidr_resources.exact_match(cidr.address);
-
-                match existing {
-                    Some(existing) => existing.id != cidr.id,
-                    None => true,
-                }
+        match &new_resource {
+            Resource::Dns(dns) => {
+                // No-op if address is already in the resolver
+                let _ = self.stub_resolver.add_resource(dns.id, dns.address.clone());
+            }
+            Resource::Cidr(_cidr) => {
+                // No-op if address is already in the table
+                self.maybe_update_cidr_resources();
             }
             Resource::Internet(resource) => {
-                self.internet_resource.replace(resource.id) != Some(resource.id)
+                // No-op if resource ID hasn't changed
+                let _ = self.internet_resource.replace(resource.id);
             }
         };
 
-        if !added {
-            return;
-        }
-
         let name = new_resource.name();
         let address = new_resource.address_string().map(tracing::field::display);
         let sites = new_resource.sites_string();
 
         tracing::info!(%name, address, %sites, "Activating resource");
```

Comment:
The reason we had this logic was to make sure this log is truthful. It needs to be in sync with what the functions above (that you commented with no-ops) actually do.

Reply:
Maybe you just want to always issue updated resources but keep the logic around whether it got added?

Reply:
> The reason we had this logic was to make sure this log is truthful.

Hmm, I must be confused then. What does `Activating` mean to a user?

Reply:
It means something to us when debugging logs!

Reply:
> Maybe you just want to always issue updated resources but keep the logic around whether it got added?

Yeah could do that too. Only log when `added`. But the emit_resources_changed always happens.

Reply:
Noted, will get this fixed!

Reply:
Also, we don't want to log this unless something changed to not spam it and be truthful. For someone with many resources, this will other log _a lot_ (once for each resource on every network reset).